src/integration/resources/options.go

```diff
@@ -10,10 +10,11 @@ import (
 type ClusterOptions struct {
 	// DBNode contains cluster options for spinning up dbnodes.
 	DBNode *DBNodeClusterOptions
-
 	// Aggregator is the optional cluster options for spinning up aggregators.
 	// If Aggregator is nil, the cluster contains only m3coordinator and dbnodes.
 	Aggregator *AggregatorClusterOptions
+	// CoordinatorGeneratePorts indicates whether to update the coordinator config to use open ports.
+	CoordinatorGeneratePorts bool
```

Comment:
This should live under a `Coordinator` specific section so probably want to add a CoordinatorClusterOptions struct